src/document/DocumentCommandHandlers.js

```diff
@@ -1718,6 +1718,20 @@ define(function (require, exports, module) {
 
     /** Reload Without Extensions commnad handler **/
     var handleReloadWithoutExts = _.partial(handleReload, true);
+    
+    /** 
+     * Attach a beforeunload handler to notify user about unsaved changes. 
+     * Prevents data loss in scenario reported under #13708
+    **/
+    window.onbeforeunload = function(e) {
```

Comment:
this will be called even in normal scenarios right? will the return value be a NOP in those cases?

Reply:
Yes. To safeguard that , there is an additional  flag check along with non zero dirty files. We will get in this handler even for a normal app close, but we will not do anything in that case.

Reply:
ok.